doc/source/admin/data.md

```diff
@@ -0,0 +1,595 @@
+:language: yaml
+```
+
+On the other hand, you might run a small lab with a dedicate MinIO storage service and just trust
+your user's to define individual buckets by name:
+
+```{literalinclude} ../../../lib/galaxy/objectstore/templates/examples/minio_just_buckets_legacy.yml
+:language: yaml
+```
+
+The syntax for the ``configuration`` section of ``generic_s3`` templates looks like this.
+
+![](object_store_generic_s3_configuration_template.png)
+
+At runtime, after the ``configuration`` template is expanded, the resulting dictionary
+passed to Galaxy's object store infrastructure looks like this and should match a subset
+of what you'd be able to add directly to ``object_store_conf.yml`` (Galaxy's global object
+store configuration).
+
+![](object_store_generic_s3_configuration.png)
+
+### YAML Syntax
+
+![galaxy.objectstore.templates.models](object_store_templates.png)
+
+### Ready To Use Production Object Store Templates
+
+The templates have been tested by a Galaxy developer and are sufficiently generic that
+they may make sense for a variety of Galaxy instances, address a variety of potential use
+cases, and do not have need any additional tailoring, parameterization, or other
+customization. These assume your Galaxy instance has a Vault configured and you're
+comfortable with it storing your user's secrets.
+
+#### Allow Users to Define Azure Blob Storage as Object Stores
+
+```{literalinclude} ../../../lib/galaxy/objectstore/templates/examples/production_azure_blob.yml
+:language: yaml
+```
+
+![Screenshot](user_object_store_form_full_azure.png)
+
+#### Allow Users to Define Generic S3 Compatible Storage Services as Object Stores
+
+```{literalinclude} ../../../lib/galaxy/objectstore/templates/examples/production_generic_s3.yml
+:language: yaml
+```
+
+#### Allow Users to Define AWS S3 Buckets as Object Stores
+
+```{literalinclude} ../../../lib/galaxy/objectstore/templates/examples/production_aws_s3.yml
+:language: yaml
+```
+
+![Screenshot](user_object_store_form_full_aws_s3.png)
+
+
+#### Allow Users to Define Google Cloud Provider S3 Interop Storage Buckets as Object Stores
+
+This template includes descriptions of how to generate HMAC keys used by this interoperability
+layer provided by Google and lots of links to relevant Google Cloud Storage documentation.
+
+```{literalinclude} ../../../lib/galaxy/objectstore/templates/examples/production_gcp_s3.yml
+:language: yaml
+```
+
+![Screenshot](user_object_store_form_full_gcp_s3_interop.png)
+
+## File Source Templates
+
+Galaxy's file source templates are configured as a YAML list of template objects. This list
+can be placed ``file_source_templates.yml`` in Galaxy configuration directory (or any path
+pointed to by the configuration option ``file_source_templates_config_file`` in ``galaxy.yml``).
+Alternatively, the  configuration can be placed directly into ``galaxy.yml`` using the
+``file_source_templates`` configuration option.
+
+
+### File Source Types
+
+#### ``posix``
```

Comment:
Is there a reason for calling this `posix` in contrast to `disk`, besides historical reasons?